test/lit/passes/optimize-instructions.wast

```diff
@@ -2687,7 +2683,9 @@
     )
   )
   ;; CHECK:      (func $sext-24-100 (result i32)
-  ;; CHECK-NEXT:  (i32.const 100)
+  ;; CHECK-NEXT:  (i32.extend8_s
+  ;; CHECK-NEXT:   (i32.const 100)
+  ;; CHECK-NEXT:  )
```

Comment:
This seems wrong.. but I have no idea.. why would this get wrapped in i32.extend8_s?

Reply:
It's ok. Some optimize instructions peephole rules may fold compound expressions to constant while single unary expression with constant input didn't touch and leave only for constant folding pass

Reply:
Maybe it would be worth further optimizing this in OptimizeInstructions even though precompute-propagate could also handle it. wdyt @kripken?

Reply:
> It's ok. Some optimize instructions peephole rules may fold compound expressions to constant (or skip some operations) while single unary expression with constant input didn't touch and leave to separate constant folding pass

I'm not sure I understand.. if the input is simply a `i32.const` why would it considered an optimization to wrap the constant in meaningless unary operation (sorry I don't know much about how these ops work).

Reply:
This input is below and it's a pair of shifts, not a constant. When sign-ext is enabled, the shifts get optimized to the extend here. When sign-ext is not enabled, optimization goes down a different path that ends up with a constant. Even though this particular optimization regressed, it's ok because precompute-propagate will later finish optimizing to a constant.

Reply:
In general I think this is fine, better to avoid redundancy between passes.

The one thing I slightly worry about is that I see the default pipeline runs precompute before the last optimize-instructions. For this particular case the opposite would have been better, but I'm not sure in general. Might be worth filing a bug to look into that, but I wouldn't block this.